subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/AbstractCrossTaskIncrementalJavaCompilationIntegrationTest.groovy

```diff
@@ -899,26 +118,21 @@ abstract class AbstractCrossTaskIncrementalJavaCompilationIntegrationTest extend
         """
 
         then:
-        fails "impl:compileJava"
+        fails "impl:${language.compileTaskName}"
         result.hasErrorOutput("package a is not visible")
     }
 
-    def "recompiles downstream dependents of classes whose package-info changed"() {
-        given:
-        def packageFile = file("api/src/main/java/foo/package-info.java")
-        packageFile.text = """package foo;"""
-        file("api/src/main/java/foo/A.java").text = "package foo; public class A {}"
-        file("api/src/main/java/bar/B.java").text = "package bar; public class B {}"
-        file("impl/src/main/java/baz/C.java").text = "package baz; import foo.A; class C extends A {}"
-        file("impl/src/main/java/baz/D.java").text = "package baz; import bar.B; class D extends B {}"
-
-        impl.snapshot { succeeds 'impl:compileJava' }
+    def "recompiles in case of conflicting changing constant values"() {
```

Comment:
This seems to be not working for Groovy compilation. Need to inspect the classfiles generated by groovy compiler.